app/src/main/java/swati4star/createpdf/activity/MainActivity.java

```diff
@@ -95,41 +94,45 @@ protected void onCreate(Bundle savedInstanceState) {
      *
      * @return - instance of current fragment
      */
-    private Fragment checkForAppShortcutClicked() {
-        Fragment fragment;
+    private Fragment checkForAppShortcutClicked(Fragment fragment) {
         FragmentManager fragmentManager = getSupportFragmentManager();
-        switch (Objects.requireNonNull(getIntent().getAction())) {
-            case ACTION_SELECT_IMAGES:
-                fragment = new ImageToPdfFragment();
-                Bundle bundle = new Bundle();
-                bundle.putBoolean(OPEN_SELECT_IMAGES, true);
-                fragment.setArguments(bundle);
-                break;
-            case ACTION_VIEW_FILES:
-                fragment = new ViewFilesFragment();
-                setDefaultMenuSelected(1);
-                break;
-            case ACTION_TEXT_TO_PDF:
-                fragment = new TextToPdfFragment();
-                setDefaultMenuSelected(4);
-                break;
-            case ACTION_MERGE_PDF:
-                fragment = new MergeFilesFragment();
-                setDefaultMenuSelected(2);
-                break;
-            default:
-                // Set default fragment
-                fragment = new HomeFragment();
-                break;
+        if (getIntent() != null && getIntent().getAction() != null ) {
```

Comment:
Why are these changes needed?

Reply:
because getIntent().getAction() returns a null pointer exception when the activity is opened, so i  make a few change to avoid the error.

Reply:
If this is the case, then you can just add a null check.